src/abi/abi_type.ts

```diff
@@ -0,0 +1,846 @@
+/* eslint-disable no-bitwise */
+/* eslint-disable no-use-before-define */
+/* eslint-disable class-methods-use-this */
+
+/**
+    //ABI-Types: uint<N>: An N-bit unsigned integer (8 <= N <= 512 and N % 8 = 0).
+    // | byte (alias for uint8)
+    // | ufixed <N> x <M> (8 <= N <= 512, N % 8 = 0, and 0 < M <= 160)
+    // | bool
+    // | address (alias for byte[32])
+    // | <type> [<N>]
+    // | <type> []
+    // | string
+    // | (T1, ..., Tn)
+*/
+import algosdk from '../..';
+import { concatArrays } from '../utils/utils';
```

Comment:
To avoid circular imports, you should do `import { encodeAddress, decodeAddress } from '../encoding/address'` instead